src/ChemicalDiffusion/ChemicalDiffusion.jl

```diff
@@ -0,0 +1,349 @@
+
+    if P isa Quantity && T isa Quantity
+        @unpack_units D0, Ea, ΔV, P0, R = data
+    else
+        @unpack_val D0, Ea, ΔV, P0, R = data
+    end
+
+    D = D0 * exp(- (Ea + (P - P0) * ΔV) / (R * T))
+
+    return D
+end
+
+"""
+    compute_D!(D, data::DiffusionData; T=ones(size(D))K, P=zeros(size(D))Pa, kwargs...)
+
+In-place version of `compute_D(data::DiffusionData; T=1K, P=0Pa, kwargs...)`. `D` should be an array of the same size as T and P.
+"""
+function compute_D!(
+    D::AbstractArray{_T,nDim},
+    data::DiffusionData;
+    T=ones(size(D))K,
+    P=zeros(size(D))Pa,
+    kwargs...) where {_T, nDim}
+
+    @inbounds for i in eachindex(D)
+        D[i] = compute_D(data; T=T[i], P=P[i], kwargs...)
+    end
+end
+
+
+function show(io::IO, g::DiffusionData)
+    return print(
+        io,
+        "DiffusionData: Mineral = $(unsafe_string(g.Mineral)), Species = $(unsafe_string(g.Species)), D0 = $(Value(g.D0)), Ea = $(Value(g.Ea)), ΔV = $(Value(g.ΔV))",
+    )
+end
+
+
+export AbstractChemicalDiffusion,
+       DiffusionData,
+       compute_D,
+       compute_D!
+
+
+"""
+    SetChemicalDiffusion["Name of Chemical Diffusion"]
+"""
+function SetChemicalDiffusion(
+    name:: F;
+    D0          = nothing,
+    D0_2σ       = nothing,
+    Ea          = nothing,
+    Ea_2σ       = nothing,
+    ΔV          = nothing,
+    ΔV_2σ       = nothing,
+    T_range_min = nothing,
+    T_range_max = nothing,
+    P0          = nothing,
+    ) where F
+    kwargs = (; D0, D0_2σ, Ea, Ea_2σ, ΔV, ΔV_2σ, T_range_min, T_range_max, P0)
+    Transform_ChemicalDiffusion(name, kwargs)
+end
+
+function SetChemicalDiffusion(
+    name::F,
+    CharDim::GeoUnits{T};
+    D0          = nothing,
+    D0_2σ       = nothing,
+    Ea          = nothing,
+    Ea_2σ       = nothing,
+    ΔV          = nothing,
+    ΔV_2σ       = nothing,
+    T_range_min = nothing,
+    T_range_max = nothing,
+    P0          = nothing,
+    ) where {F, T<:Union{GEO, SI}}
+    kwargs = (; D0, D0_2σ, Ea, Ea_2σ, ΔV, ΔV_2σ, T_range_min, T_range_max, P0)
+    nondimensionalize(Transform_ChemicalDiffusion(name, kwargs), CharDim)
+end
+
+"""
+    Transform_ChemicalDiffusion(name)
+Transforms units from MPa, kJ etc. to basic units such as Pa, J etc.
+"""
+Transform_ChemicalDiffusion(name::F) where F = Transform_ChemicalDiffusion(diffusion_database(name))
+Transform_ChemicalDiffusion(name::F, kwargs::NamedTuple) where F = Transform_ChemicalDiffusion(diffusion_database(name), kwargs)
+
+function Transform_ChemicalDiffusion(name::F, CharDim::GeoUnits{U}) where {F, U<:Union{GEO,SI}}
+    Transform_ChemicalDiffusion(diffusion_database(name), CharDim)
+end
+
+function Transform_ChemicalDiffusion(p::AbstractChemicalDiffusion{T}, CharDim::GeoUnits{U}) where {T,U<:Union{GEO,SI}}
+    nondimensionalize(Transform_ChemicalDiffusion(p), CharDim)
+end
+
+function Transform_ChemicalDiffusion(pp::AbstractChemicalDiffusion{T}) where T
```

Comment:
```suggestion
function Transform_ChemicalDiffusion(pp::AbstractChemicalDiffusion)
```
You don't use `T`, so you can just nod add it.